### ISSUE TITLE
ThreeDS2DataBuilder Class is broken

### ISSUE THREAD

**Issue by paolibbe** [Adyen/adyen-magento2#571]
**Magento version**: 2.3.2
**Plugin version**: 5.0.0
**Description**
Trying to install version 5.0.0, i found that the class placed in **Adyen/Payment/Gateway/Request/ThreeDS2DataBuilder.php**
has a wrong naming, as shown below. This mismatching is causing issues with setup:di:compile command and checkout stop working with this error.

```
<?php
/**
 *                       ######
 *                       ######
 * ############    ####( ######  #####. ######  ############   ############
 * #############  #####( ######  #####. ######  #############  #############
 *        ######  #####( ######  #####. ######  #####  ######  #####  ######
 * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
 * ###### ######  #####( ######  #####. ######  #####          #####  ######
 * #############  #############  #############  #############  #####  ######
 *  ############   ############  #############   ############  #####  ######
 *                                      ######
 *                               #############
 *                               ############
 *
 * Adyen Payment module (https://www.adyen.com/)
 *
 * Copyright (c) 2015 Adyen BV (https://www.adyen.com/)
 * See LICENSE.txt for license details.
 *
 * Author: Adyen <magento@adyen.com>
 */

namespace Adyen\Payment\Gateway\Request;

use Magento\Payment\Gateway\Request\BuilderInterface;

class RecurringDataBuilder implements BuilderInterface
{
    /**
     * @var \Magento\Framework\App\State
     */
    private $appState;

    /**
     * @var \Adyen\Payment\Helper\Requests
     */
    private $adyenRequestsHelper;

    /**
     * RecurringDataBuilder constructor.
     *
     * @param \Magento\Framework\Model\Context $context
     * @param \Adyen\Payment\Helper\Requests $adyenRequestsHelper
     */
    public function __construct(
        \Magento\Framework\Model\Context $context,
        \Adyen\Payment\Helper\Requests $adyenRequestsHelper
    ) {
        $this->appState = $context->getAppState();
        $this->adyenRequestsHelper = $adyenRequestsHelper;
    }


    /**
     * @param array $buildSubject
     * @return array
     * @throws \Magento\Framework\Exception\LocalizedException
     */
    public function build(array $buildSubject)
    {
        /** @var \Magento\Payment\Gateway\Data\PaymentDataObject $paymentDataObject */
        $paymentDataObject = \Magento\Payment\Gateway\Helper\SubjectReader::readPayment($buildSubject);
        $payment = $paymentDataObject->getPayment();
        $storeId = $payment->getOrder()->getStoreId();
        $areaCode = $this->appState->getAreaCode();

        return $this->adyenRequestsHelper->buildRecurringData([], $areaCode, $storeId, $payment);
    }
}
```

Another minor issue is that module version, inside the module.xml file, is still on 4.5.4, could you please upgrade this one too, just to clarify in logs which version is used?

Thank you,
Paolo

**Comment by cyattilakiss** [comment]
Hi @paolibbe ,

Thank you for raising the issue. We are going to release this referenced fix #572 soon.

Best,
Attila
Adyen

**Comment by cyattilakiss** [comment]
Hi @paolibbe ,

The fix has been released.

Best,
Attila
Adyen
